tlsconfig/tlsconfig_server.go

```diff
@@ -26,7 +26,7 @@ func NewServerConfig(tlsCertProvider TLSCertProvider, params ...ServerParam) (*t
 	if tlsCertProvider == nil {
 		return nil, fmt.Errorf("tlsCertProvider provided to NewServerConfig was nil")
 	}
-	configurers := []configurer{authKeyPairParam(tlsCertProvider)}
+	configurers := []configurer{certificatesParam(tlsCertProvider)}
```

Comment:
Wanted to make a parallel change for servers too because Golang TLS conf has a similar `GetCertificate` func, however ran into an issue that stems from this logic that determines if `GetCertificate` gets called:
```
	if c.GetCertificate != nil &&
		(len(c.Certificates) == 0 || len(clientHello.ServerName) > 0) {
		cert, err := c.GetCertificate(clientHello)
		if cert != nil || err != nil {
			return cert, err
		}
	}
```

As an example, [tlsconfig_test](https://github.com/palantir/pkg/blob/master/tlsconfig/tlsconfig_test.go) doesn't set a servername, and the `StartTLS()` func adds a local host cert when it detects that the certificates list is empty. This means the tests fails as the local host cert isn't recognized by the client. It's possible that this edge case could happen in the wild as well

Don't want to block this PR so I'm scoping this PR to client behavior specifically, but worth considering the server side as a FLUP